pushy/src/main/java/com/turo/pushy/apns/proxy/HttpProxyHandlerFactory.java

```diff
@@ -41,6 +47,53 @@
     private final String username;
     private final String password;
 
+    /**
+     * Checks the system default proxies to determine if an {@link HttpProxyHandlerFactory}
+     * is required for the given host or not, returning an instance or null for each case.
+     *
+     * @param apnsHost the APNs host to check e.g. "api.push.apple.com".
+     * @return an {@link HttpProxyHandlerFactory} if one is required, or <code>null</code> if not.
+     * @throws URISyntaxException if <code>apnsHost</code> is malformed in some way.
+     *
+     * @see #fromSystemProxies()
+     * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#Proxies">The Java documentation on proxies.</a>
+     * @since 0.14
+     */
+    public static HttpProxyHandlerFactory fromSystemProxies(final String apnsHost)
+            throws URISyntaxException {
+
+        URI hostUri = new URI("https", apnsHost, null, null);
+        SocketAddress proxyAddress = ProxyLocator.getProxyAddressForUri(hostUri, Proxy.Type.HTTP);
+
+        if (proxyAddress != null) {
```

Comment:
Minor style thing: we prefer single `return` statements where possible. Could we turn this into a ternary instead (assuming we want to continue returning `null` instead of throwing an exception)?

```java
return proxyAddress != null ? new HttpProxyHandlerFactory(proxyAddress) : null;
```

Reply:
Three cheers for the ternary!